integration/command.go

```diff
@@ -198,44 +214,146 @@ func (c *command) verifyOutput() error {
-
+	// Given that we set Setpgid, here we just need to send the PID of /bin/sh
+	// (which is the same PGID) as a negative number to syscall.Kill(). As a
+	// result, the signal will be received by all the processes with such PGID,
+	// in our case, the process of /bin/sh and c.cmd.
+	err := syscall.Kill(-c.command.Process.Pid, sig)
 	if err != nil {
-		t.Fatal(err)
+		return err
 	}
 
-	err = c.verifyOutput()
-	if err != nil {
-		t.Fatal(err)
+	// In some cases, we do not have to wait here because the cmd was executed
+	// with Run(), which already waits. On the contrary, in the case it was
+	// executed with Start() thus c.started is true, we need to wait indeed.
+	if c.started {
+		err = c.command.Wait()
+		if err == nil {
+			return nil
+		}
+
+		// Verify if the error is about the signal we just sent. In that case,
+		// do not return error, it is what we were expecting.
+		var exiterr *exec.ExitError
+		if ok := errors.As(err, &exiterr); !ok {
+			return err
+		}
+
+		waitStatus, ok := exiterr.Sys().(syscall.WaitStatus)
+		if !ok {
+			return err
+		}
+
+		if waitStatus.Signal() != sig {
+			return err
+		}
+
+		return nil
 	}
+
+	return err
 }
 
 // runWithoutTest runs the command, this is thought to be used in TestMain().
 func (c *command) runWithoutTest() error {
+	c.createExecCmd()
+
 	fmt.Printf("Run command: %s\n", c.cmd)
+	err := c.command.Run()
+	fmt.Printf("Command returned:\n%s\n%s\n", c.stderr.String(), c.stdout.String())
+
+	if err != nil {
+		return err
+	}
+
+	return c.verifyOutput()
+}
+
+// startWithoutTest starts the command, this is thought to be used in TestMain().
+func (c *command) startWithoutTest() error {
+	if c.started {
+		fmt.Printf("Warn: trying to start a command but it is not running: %s\n", c.cmd)
+		return nil
+	}
 
 	c.createExecCmd()
-	err := c.command.Run()
 
+	fmt.Printf("Start command: %s\n", c.cmd)
+	err := c.command.Start()
+	if err != nil {
+		return err
+	}
+
+	c.started = true
+
+	return nil
+}
+
+// waitWithoutTest waits for a command that was started with startWithoutTest(),
+// this is thought to be used in TestMain().
+func (c *command) waitWithoutTest() error {
+	if !c.started {
+		fmt.Printf("Warn: trying to wait for a command that has not been started yet: %s\n", c.cmd)
+		return nil
+	}
+
+	fmt.Printf("Wait for command: %s\n", c.cmd)
+	err := c.command.Wait()
 	fmt.Printf("Command returned:\n%s\n%s\n", c.stderr.String(), c.stdout.String())
 
 	if err != nil {
 		return err
 	}
 
-	return c.verifyOutput()
+	c.started = false
+
+	return nil
+}
+
+// killWithoutTest kills for a command that was started with startWithoutTest()
+// or runWithoutTest() and we do not need to verify its output. This is thought
+// to be used in TestMain().
+func (c *command) killWithoutTest() error {
+	fmt.Printf("Kill command(%s)\n", c.name)
+
+	if err := c.kill(); err != nil {
+		return fmt.Errorf("failed to kill command(%s): %w", c.name, err)
+	}
+
+	return nil
+}
```

Comment:
How is it different than just calling `c.kill()`, is it all about the extra logging information that is printed or are there more differences?

Reply:
The extra logging and for keeping all the commands align with the naming `{start|wait|run|kill}WithoutTest()`